harness/determined/callback.py

```diff
@@ -10,11 +10,19 @@ def on_trial_begin(self) -> None:
         """Executed before the start of the first training step of a trial."""
         pass
 
-    def on_train_step_begin(self, step_id: int) -> None:
+    def on_train_step_begin(
```

Comment:
Totally optional: this Callback struct is purely historical (I'm like 90% sure).  I wouldn't be opposed to deleting it and just calling the Tensorboard BatchWriter directly, as opposed to updating this interface just for the sake of that one entirely internal implementation of this interface.

But there's not a need to eliminate this object at the moment either, so feel free to update it and ignore it.

Reply:
I'll leave the callbacks incase they're useful in the future and because I'm partial to [AOP](https://en.wikipedia.org/wiki/Aspect-oriented_programming); writing tensor board metrics and the control flow handled by the WorkloadManager seem like cross cutting concerns that are separated them nicely by injecting callbacks (to me).